### PR TITLE
chore: move coverage report into a separate job

### PR DESCRIPTION
**Pull request by tatomyr**
## What/Why/How?

Fixes github action failing on MR from forks due to lacking permissions by splitting coverage report into a separate always-passing job.

## Check yourself

- [ ] Code is linted
- [ ] Tested with redoc/reference-docs/workflows
- [ ] All new/updated code is covered with tests

## Security

- [x] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -32,15 +32,32 @@ jobs:
...
+          key: npm-${{ hashFiles('package-lock.json') }}
+          restore-keys: |
+            npm-${{ hashFiles('package-lock.json') }}
+            npm-
+      - name: Install dependencies
+        run: npm ci
+        env:
+          CI: true
       - name: Coverage Report
         uses: artiomtr/jest-coverage-report-action@v2.0-rc.4
```
Comment (roman-sainchuk) [inline .github/workflows/tests.yaml:58]:
```suggestion
        uses: artiomtr/jest-coverage-report-action@v2.0.8
```

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -32,15 +32,32 @@ jobs:
...
+      - uses: actions/checkout@v3
+      - name: Cache dependencies
+        uses: actions/cache@v3
+        with:
+          path: ~/.npm
+          key: npm-${{ hashFiles('package-lock.json') }}
+          restore-keys: |
+            npm-${{ hashFiles('package-lock.json') }}
+            npm-
+      - name: Install dependencies
```
Comment (roman-sainchuk) [inline .github/workflows/tests.yaml:53]: We can remove `Install dependencies` step as `Coverage report` step does it

File: .github/workflows/tests.yaml [hunk 3]
```diff
@@ -32,15 +32,32 @@ jobs:
...
+        run: npm ci
+        env:
+          CI: true
       - name: Coverage Report
         uses: artiomtr/jest-coverage-report-action@v2.0-rc.4
+        continue-on-error: true
```
Comment (roman-sainchuk) [inline .github/workflows/tests.yaml:59]: Cannot find such an option for this action, does it really exists ?

File: .github/workflows/tests.yaml [hunk 4]
```diff
@@ -32,15 +32,32 @@ jobs:
...
         with:
           skip-step: none
-          coverage-file: report-incoming.json
           annotations: none
           test-script: npm run jest -- --silent --ci --json --testLocationInResults --outputFile=report.json --bail --coverage
```
Comment (roman-sainchuk) [inline .github/workflows/tests.yaml:63]:
```suggestion
          test-script: npm run jest
```

**Review by roman-sainchuk — CHANGES_REQUESTED** [review]

File: .github/workflows/tests.yaml (hunk 3 above)
Comment (tatomyr) [reply .github/workflows/tests.yaml:59]: Here is the [docs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idcontinue-on-error).

Comment (roman-sainchuk) [reply .github/workflows/tests.yaml:59]: Wow, thank you

File: .github/workflows/tests.yaml [hunk 5]
```diff
@@ -32,15 +32,32 @@ jobs:
...
+        with:
+          path: ~/.npm
+          key: npm-${{ hashFiles('package-lock.json') }}
+          restore-keys: |
+            npm-${{ hashFiles('package-lock.json') }}
+            npm-
+      - name: Install dependencies
+        run: npm ci
+        env:
+          CI: true
```
Comment (roman-sainchuk) [inline .github/workflows/tests.yaml:56]:
```suggestion
```

**Review by roman-sainchuk — APPROVED** [review]

**Pull request merged**